### ISSUE TITLE
Pyright reports reportAttributeAccessIssue when importing glob.fnmatch

### ISSUE THREAD

**Issue by attilaolah** [microsoft/pyright#10136]
**Describe the bug**

Running `pyright test.py` against a small test file (contents below) produces this error:

```
/tmp/pyright/test.py
  /tmp/pyright/test.py:1:18 - error: "fnmatch" is unknown import symbol (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations 
```

I silence the error in my projects by creating a type stub for `glob.fnmatch`, but:

1. This repo [already contains a type stub](https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/typeshed-fallback/stdlib/fnmatch.pyi) for `glob.fnmatch`, and
2. even with that workaround, I get a warning:
    ```
    warning: Import "glob.fnmatch" could not be resolved from source (reportMissingModuleSource)
    ```

**Code or Screenshots**

```python
from glob import fnmatch

print(fnmatch.fnmatch("a", "*"))
```

**VS Code extension or command-line**

Command line, version 1.1.397, issue also present in a few earlier versions I tried, as well as when running as an LSP with Neovim.

**Comment by attilaolah** [comment]
Looks like I somehow thought `fnmatch` was exported from `glob`; pardon my ignorance.
